vllm/entrypoints/openai/serving_chat.py

```diff
@@ -112,39 +112,37 @@ def _parse_chat_message_content_parts(
             part_type = part["type"]
             if part_type == "text":
                 text = cast(ChatCompletionContentPartTextParam, part)["text"]
-
                 texts.append(text)
             elif part_type == "image_url":
                 if vlm_config is None:
                     raise ValueError(
                         "'image_url' input is not supported as the loaded "
                         "model is not multimodal.")
+                assert self.tokenizer is not None
+                image_url = cast(ChatCompletionContentPartImageParam,
+                                 part)["image_url"]
 
-                elif len(image_futures) == 0:
-                    assert self.tokenizer is not None
-                    image_url = cast(ChatCompletionContentPartImageParam,
-                                     part)["image_url"]
-
-                    if image_url.get("detail", "auto") != "auto":
-                        logger.warning(
-                            "'image_url.detail' is currently not supported and "
-                            "will be ignored.")
+                if image_url.get("detail", "auto") != "auto":
+                    logger.warning(
+                        "'image_url.detail' is currently not supported and "
+                        "will be ignored.")
 
-                    image_future = async_get_and_parse_image(image_url["url"])
-                    image_futures.append(image_future)
+                async def async_get_and_parse_image(image_url: str):
+                    with await ImageFetchAiohttp.fetch_image(image_url
```

Comment:
Since you call `load` inside `fetch_image`, we don't need to use a context manager anymore.

Reply:
yes, yes removing the ctx manager.